BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeGraphQLClient.kt

```diff
@@ -0,0 +1,81 @@
+package com.braintreepayments.api
+
+internal class BraintreeGraphQLClient(
+    private val httpClient: HttpClient = createDefaultHttpClient()
+) {
+
+    fun post(
+        path: String?,
```

Comment:
Since this is internal should we consider making some of the parameters (like path) required?

Reply:
Yeah we will eventually, but Java's lack of support for optionals makes things complicated. We're planning on doing a second pass at removing optionals and improving the Kotlin code quality once everything has been converted.